### PR TITLE
project: add resource for subdivisions

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Adds configuration to manage subdivisions in administration.
* Creates a brief view to display subdivisions items.
* Creates a detail view to display a subdivision.
* Adds a link to subdivisions in admin menu.
* Reorganizes the admin navigation.
* Adds facet for `subdivisions` in documents.
* Adds facet for `subdivisions` in deposits.
* Adds facet for `subdivisions` in users.
* Displays subdivisions info in document detail view in admin.
* Displays subdivisions info in the user detail view.
* Displays organisations and subdivisions in document brief view.
* Displays organisations and subdivisions in document detail view below the title.
* Displays subdivisions and collections in organisation detail view.
* Displays subdivisions in user brief view.
* Enables editor long mode for users.
* Checks the existence of the link to detail in documents brief view to avoid an error.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

File: projects/sonar/src/app/record/organisation/detail/detail.component.html [hunk 1]
```diff
@@ -43,12 +43,39 @@ <h1 class="mb-4">{{ record.metadata.name }}</h1>
         <dt class="col-sm-3">{{ 'Custom field' | translate }} {{ i }}</dt>
         <dd class="col-sm-9">
           <ng-container *ngIf="record.metadata['documentsCustomField' + i].label">
-            {{ 'Label' | translate }}: {{ record.metadata['documentsCustomField' + i].label | languageValue | async }}<br>
+            {{ 'Label' | translate }}: {{ record.metadata['documentsCustomField' + i].label | languageValue | async
+            }}<br>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/organisation/detail/detail.component.html:47]: Put the closure on the same line

File: projects/sonar/src/app/record/document/document.component.html [hunk 2]
```diff
@@ -16,25 +16,54 @@
...
-      <ng-template #routingLink>
-        <a [routerLink]="detailUrl.link">
-          <span [innerHTML]="record.metadata.title[0].mainTitle | languageValue | async | nl2br"></span><ng-container
-            *ngIf="record.metadata.title[0].subtitle">:&nbsp;{{ record.metadata.title[0].subtitle | languageValue | async }}
-          </ng-container>
+      <ng-container *ngIf="detailUrl; else titleWithoutLink">
+        <a [href]="detailUrl.link" *ngIf="detailUrl.external; else routingLink">
+          <span [innerHTML]="record.metadata.title[0].mainTitle | languageValue | async | nl2br"></span>
+          <ng-container *ngIf="record.metadata.title[0].subtitle">
+            :&nbsp;{{ record.metadata.title[0].subtitle | languageValue | async }}</ng-container>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/record/document/document.component.html:37]: new line before `</ng-container>`

File: projects/sonar/src/app/_layout/admin/admin.component.html [hunk 3]
```diff
@@ -25,22 +25,21 @@
...
                 aria-haspopup="true" aria-expanded="false" dropdownToggle translate>
-                Documents
+                Administration
               </a>
               <div class="dropdown-menu dropdown-menu-right" *dropdownMenu>
-                <a class="dropdown-item" routerLink="/records/documents" translate>Documents</a>
                 <a class="dropdown-item" routerLink="/records/collections" translate>Collections</a>
+                <a class="dropdown-item" routerLink="/records/organisations" translate>Organisations</a>
+                <a class="dropdown-item" routerLink="/records/subdivisions" translate>Subdivisions</a>
+                <a class="dropdown-item" routerLink="/records/users" translate>Users</a>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/_layout/admin/admin.component.html:37]: use an array to define the routerlink (same has on detail.component.html file)

File: projects/sonar/src/app/_layout/admin/admin.component.html [hunk 4]
```diff
@@ -25,22 +25,21 @@
...
               </div>
             </li>
+            <li class="nav-item" *ngIf="user.is_moderator">
+              <a class="nav-link" routerLink="/records/documents" translate>Documents</a>
```
Comment (Garfield-fr) [inline projects/sonar/src/app/_layout/admin/admin.component.html:41]: same here

**Review by Garfield-fr — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

**Pull request merged**